pkg/apis/core/validation/validation.go

```diff
@@ -6407,7 +6407,9 @@ func ValidateSecret(secret *core.Secret) field.ErrorList {
 func ValidateSecretUpdate(newSecret, oldSecret *core.Secret) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newSecret.ObjectMeta, &oldSecret.ObjectMeta, field.NewPath("metadata"))
 
-	allErrs = append(allErrs, ValidateImmutableField(newSecret.Type, oldSecret.Type, field.NewPath("type"))...)
+	if oldSecret.Namespace != "openshift-kube-apiserver-operator" && oldSecret.Namespace != "openshift-kube-apiserver" {
```

Comment:
it looks like we need to patch https://github.com/openshift/library-go/blob/2c99390d34b0fd086758f66035a768191a67a1e9/pkg/operator/resource/resourceapply/core.go#L429 as well - to stop deletion.

Reply:
in https://github.com/openshift/library-go/pull/1703

Reply:
@deads2k why do you want to also relax the validation for `openshift-kube-apiserver` ?